<?xml version="1.0"?>
<launch>
    <node pkg="rosbag" type="record" name="rosbag_record"
            args="-o $(find softjig_ros)/bag/grasp_exp
                /external_camera/color/image_raw/compressed
                /hand_camera/color/image_raw/compressed
                /gripper/output
                /iiwa/state/JointTorque"/>

    <include file="$(find pigpio_controller)/launch/remote.launch"/>
    <node pkg="rosservice" type="rosservice" name="softjig_free" args="call --wait /control_pin true"/>

    <arg name="gripper_name" default="gripper"/>
    <arg name="min_gap" default="0.0"/>
    <arg name="max_gap" default="0.085"/>
    <arg name="min_effort" default="0.0"/>
    <arg name="max_effort" default="100.0"/>

    <node pkg="robotiq_2f_gripper_control" type="Robotiq2FGripperRtuNode.py" name="robotiq_2f_gripper_rtu_node" args="/dev/ttyUSB0" output="screen">
        <remap from="Robotiq2FGripperRobotInput" to="$(arg gripper_name)/input"/>
        <remap from="Robotiq2FGripperRobotOutput" to="$(arg gripper_name)/output"/>
    </node>
    <node name="robotiq_2f_gripper_action_server" pkg="robotiq_2f_gripper_action_server" type="robotiq_2f_gripper_action_server_node">
        <param name="gripper_name" type="str" value="$(arg gripper_name)" />
        <param name="min_gap" type="double" value="$(arg min_gap)"/>
        <param name="max_gap" type="double" value="$(arg max_gap)"/>
        <param name="min_effort" type="double" value="$(arg min_effort)"/>
        <param name="max_effort" type="double" value="$(arg max_effort)"/>
        <remap from="input" to="$(arg gripper_name)/input"/>
        <remap from="output" to="$(arg gripper_name)/output"/>
    </node>
    <node pkg="rostopic" type="rostopic" name="open_gripper" args="pub /gripper/output robotiq_2f_gripper_control/Robotiq2FGripper_robot_output '{rACT: 1, rGTO: 1, rATR: 0, rPR: 0, rSP: 100, rFR: 100}' -1"/>

    <node pkg="softjig_ros" type="softjig_controller_node" name="softjig_controller_node" respawn="false" output="screen">
        <rosparam param="initial_joints">[-1.036, 1.735, -1.738, 0.968, -0.054, -1.619, -1.801]</rosparam>
    </node>
</launch>